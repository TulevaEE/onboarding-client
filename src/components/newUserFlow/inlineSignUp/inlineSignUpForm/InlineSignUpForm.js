import React from 'react';
import { PropTypes as Types } from 'prop-types';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';
import { FormattedMessage, useIntl } from 'react-intl';
import { emailValidator, requiredField } from '../../../common/form';

const renderField = ({ input, type, placeholder, disabled, meta: { touched, error } }) => (
  <div>
    <div className={`mb-3 ${touched && error ? 'has-error' : ''}`}>
      <input
        {...input}
        type={type}
        placeholder={placeholder}
        disabled={disabled}
        className="form-control"
      />
      {touched && error && (
        <div className="help-block">
          <FormattedMessage id={`new.user.flow.signup.error.${error}`} />
        </div>
      )}
    </div>
  </div>
);

export const InlineSignUpForm = ({ handleSubmit, invalid, submitting, error }) => {
  const { formatMessage } = useIntl();

  return (
    <div>
      <form id="register-form" onSubmit={handleSubmit}>
        <div className="mb-2">
          <label htmlFor="register-form-email">
            <FormattedMessage id="new.user.flow.signup.email" />
          </label>
          <Field
            component={renderField}
            type="email"
            name="email"
            id="register-form-email"
            placeholder={formatMessage({ id: 'new.user.flow.signup.email' })}
            validate={[requiredField, emailValidator]}
          />
        </div>
        <div className="mb-2">
          <label htmlFor="register-form-phoneNumber">
            <FormattedMessage id="new.user.flow.signup.phoneNumber" />
          </label>
          <Field
            component={renderField}
            type="text"
            name="phoneNumber"
            id="register-form-phoneNumber"
            placeholder={formatMessage({ id: 'new.user.flow.signup.phoneNumber' })}
          />
        </div>
        <div className="mb-2">
          <span className="custom-control-indicator" />
          <div className="custom-control-description">
            <FormattedMessage id="new.user.flow.signup.tos.start" />
            <a
              href="https://tuleva.ee/wp-content/uploads/2017/10/P%C3%B5hikiri-Tulundus%C3%BChistu-Tuleva.pdf"
              target="_blank"
              rel="noopener noreferrer"
            >
              <FormattedMessage id="new.user.flow.signup.tos.statute" />
            </a>
            <FormattedMessage id="new.user.flow.signup.tos.end" />
          </div>
        </div>
        <div>
          {error && (
            <div className="alert alert-danger mb-2" role="alert">
              <FormattedMessage id={`new.user.flow.signup.error.${error}`} />
            </div>
          )}
          <button
            type="submit"
            disabled={invalid || submitting}
            className="btn btn-primary btn-lg btn-block"
          >
            <FormattedMessage id="new.user.flow.signup.submit" />
          </button>
        </div>
      </form>
    </div>
  );
};

const noop = () => null;

InlineSignUpForm.defaultProps = {
  handleSubmit: noop,
  invalid: true,
  submitting: false,
  error: '',
};

InlineSignUpForm.propTypes = {
  handleSubmit: Types.func,
  invalid: Types.bool,
  submitting: Types.bool,
  error: Types.string,
};

const mapStateToProps = (state) => ({
  initialValues: state.login.user ? { ...state.login.user } : null,
});

export default connect(mapStateToProps, null)(reduxForm({ form: 'signUp' })(InlineSignUpForm));
