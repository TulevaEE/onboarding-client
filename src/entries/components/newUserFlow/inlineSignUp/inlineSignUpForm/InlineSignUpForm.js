/* eslint-disable react/prop-types */

import React from 'react';
import { PropTypes as Types } from 'prop-types';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';
import { Message, withTranslations } from 'retranslate';
import { requiredField, length11, emailValidator } from '../../../common/form';

const renderField = ({ input, type, placeholder, disabled, meta: { touched, error } }) => (
  <div>
    <div className={`form-group ${touched && error ? 'has-error' : ''}`}>
      <input
        {...input}
        type={type}
        placeholder={placeholder}
        disabled={disabled}
        className="form-control"
      />
      {touched && error && (
        <div className="help-block">
          <Message>{`new.user.flow.signup.error.${error}`}</Message>
        </div>
      )}
    </div>
  </div>
);

export const InlineSignUpForm = ({
  handleSubmit,
  invalid,
  submitting,
  error,
  translations: { translate },
}) => (
  <div>
    <form id="register-form" onSubmit={handleSubmit}>
      <div className="form-group mb-2">
        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
        <label htmlFor="register-form-email">
          <Message>new.user.flow.signup.email</Message>
        </label>
        <Field
          component={renderField}
          type="email"
          name="email"
          id="register-form-email"
          placeholder={translate('new.user.flow.signup.email')}
          validate={[requiredField, emailValidator]}
        />
      </div>
      <div className="form-group mb-2">
        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
        <label htmlFor="register-form-personalCode">
          <Message>new.user.flow.signup.personalCode</Message>
        </label>
        <Field
          component={renderField}
          type="text"
          name="personalCode"
          id="register-form-personalCode"
          placeholder={translate('new.user.flow.signup.personalCode')}
          validate={[requiredField, length11]}
        />
      </div>
      <div className="form-group mb-2">
        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
        <label htmlFor="register-form-phoneNumber">
          <Message>new.user.flow.signup.phoneNumber</Message>
        </label>
        <Field
          component={renderField}
          type="text"
          name="phoneNumber"
          id="register-form-phoneNumber"
          placeholder={translate('new.user.flow.signup.phoneNumber')}
        />
      </div>
      <div className="form-check checkbox mb-2">
        <span className="custom-control-indicator" />
        <div className="custom-control-description">
          <Message>new.user.flow.signup.tos.start</Message>
          <a
            href="https://tuleva.ee/wp-content/uploads/2017/10/P%C3%B5hikiri-Tulundus%C3%BChistu-Tuleva.pdf"
            target="_blank"
            rel="noopener noreferrer"
          >
            <Message>new.user.flow.signup.tos.statute</Message>
          </a>
          <Message>new.user.flow.signup.tos.end</Message>
        </div>
      </div>
      <div>
        {error && (
          <div className="alert alert-danger mb-2" role="alert">
            <Message>{`new.user.flow.signup.error.${error}`}</Message>
          </div>
        )}
        <button
          type="submit"
          disabled={invalid || submitting}
          className="btn btn-primary btn-lg btn-block"
        >
          <Message>new.user.flow.signup.submit</Message>
        </button>
      </div>
    </form>
  </div>
);

const noop = () => null;

InlineSignUpForm.defaultProps = {
  handleSubmit: noop,
  invalid: true,
  submitting: false,
  error: '',
};

InlineSignUpForm.propTypes = {
  handleSubmit: Types.func,
  invalid: Types.bool,
  submitting: Types.bool,
  error: Types.string,
  translations: Types.shape({ translate: Types.func.isRequired }).isRequired,
};

const reduxInlineSignUpForm = reduxForm({ form: 'signUp' })(InlineSignUpForm);
const translatedForm = withTranslations(reduxInlineSignUpForm);

const mapStateToProps = state => ({
  initialValues: state.login.user ? { ...state.login.user } : null,
});

const connectToRedux = connect(
  mapStateToProps,
  null,
);

const prefilledForm = connectToRedux(translatedForm);

export default prefilledForm;
