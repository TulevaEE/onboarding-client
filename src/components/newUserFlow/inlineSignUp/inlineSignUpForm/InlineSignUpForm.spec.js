import React from 'react';
import { shallow } from 'enzyme';

import { InlineSignUpForm } from './InlineSignUpForm';

describe('InlineSignUpForm', () => {
  let component;
  let props;

  beforeEach(() => {
    props = { translations: { translate: () => '' } };
    component = shallow(<InlineSignUpForm {...props} />);
  });

  it('renders component', () => {
    expect(component);
  });

  it('has correct statute link', () => {
    expect(
      component.contains(
        '<a href="https://tuleva.ee/wp-content/uploads/2017/10/P%C3%B5hikiri-Tulundus%C3%BChistu-Tuleva.pdf"',
      ),
    );
  });
});
