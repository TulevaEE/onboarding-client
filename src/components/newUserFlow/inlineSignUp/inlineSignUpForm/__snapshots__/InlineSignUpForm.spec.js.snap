// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineSignUpForm renders component 1`] = `
<div>
  <form
    id="register-form"
    onSubmit={[Function]}
  >
    <div
      className="form-group mb-2"
    >
      <label
        htmlFor="register-form-email"
      >
        <MemoizedFormattedMessage
          id="new.user.flow.signup.email"
        />
      </label>
      <Field
        component={[Function]}
        id="register-form-email"
        name="email"
        placeholder="Email address*"
        type="email"
        validate={
          Array [
            [Function],
            [Function],
          ]
        }
      />
    </div>
    <div
      className="form-group mb-2"
    >
      <label
        htmlFor="register-form-phoneNumber"
      >
        <MemoizedFormattedMessage
          id="new.user.flow.signup.phoneNumber"
        />
      </label>
      <Field
        component={[Function]}
        id="register-form-phoneNumber"
        name="phoneNumber"
        placeholder="Phone number"
        type="text"
      />
    </div>
    <div
      className="mb-2"
    >
      <span
        className="custom-control-indicator"
      />
      <div
        className="custom-control-description"
      >
        <MemoizedFormattedMessage
          id="new.user.flow.signup.tos.start"
        />
        <a
          href="https://tuleva.ee/wp-content/uploads/2017/10/P%C3%B5hikiri-Tulundus%C3%BChistu-Tuleva.pdf"
          rel="noopener noreferrer"
          target="_blank"
        >
          <MemoizedFormattedMessage
            id="new.user.flow.signup.tos.statute"
          />
        </a>
        <MemoizedFormattedMessage
          id="new.user.flow.signup.tos.end"
        />
      </div>
    </div>
    <div>
      <button
        className="btn btn-primary btn-lg btn-block"
        disabled={true}
        type="submit"
      >
        <MemoizedFormattedMessage
          id="new.user.flow.signup.submit"
        />
      </button>
    </div>
  </form>
</div>
`;
